packages/dx-react-grid/docs/guides/filtering.md

```diff
@@ -11,33 +11,37 @@ The following plugins implement filtering features:
 
 Note that [plugin order](../README.md#plugin-order) is important.
 
-## Set up Local Filtering with the Filter Row
+## Setting up Local Filtering
 
 Import the plugins listed above to set up a Grid with basic filtering.
 
-## Uncontrolled Mode
+### Uncontrolled Mode
 
 In the [uncontrolled mode](controlled-and-uncontrolled-modes.md), specify the initial filtering conditions in the `FilteringState` plugin's `defaultFilters` property.
 
 .embedded-demo(filtering/local-filter-row)
 
-## Controlled Mode
+### Controlled Mode
 
-In the [controlled mode](controlled-and-uncontrolled-modes.md), pass the filtering options to the `FilteringState` plugin's `filters` property and handle the `onFiltersChange` event to control the filterings state externally.
+In the [controlled mode](controlled-and-uncontrolled-modes.md), pass the filtering options to the `FilteringState` plugin's `filters` property and handle the `onFiltersChange` event to control the filtering state externally.
 
 .embedded-demo(filtering/local-filtering-controlled)
 
-## Filter Row Customization
+### Using Custom Filtering Algorithms
 
-Define a filter row cell template using the `TableFilterRow` plugin's `filterCellTemplate` property to use a custom editor instead of the built-in one. Pass a function that renders a custom component to the `filterCellTemplate` property. Define the component's state and handle its changes using the `filter` and `setFilter` arguments respectively to delegate the component's state control to the `TableFilterRow` plugin. In this case, the `FilteringState` plugin handles changes internally, which allows you to use the Grid in both controlled and uncontrolled modes.
+You can also specify a filtering predicate using the `LocalFiltering` plugin's `filterFn` property to implement a custom filtering logic.
 
-## Custom Filtering Algorithm
+Alternatively, it is possible to go along without the `LocalFiltering` plugin. In this case, you should filter rows manually on the `FilteringState` plugin's `onFiltersChange` event and then pass the result to the Grid component's `row` property.
```

Comment:
It's not quite clear from the description when I should use filterFn and when I should skip using the LocalFiltering plugin.